lib/manifest.go

```
@@ -0,0 +1,381 @@
+}
+func (a *ACBuild) RemoveDependency(imageName string) (err error) {
+	if err = a.lock(); err != nil {
+		return err
+	}
+	defer func() {
+		if err1 := a.unlock(); err == nil {
+			err = err1
+		}
+	}()
+	switch m := a.man.(type) {
+	case *appc.Manifest:
+		return m.RemoveDependency(imageName)
+	}
+	return fmt.Errorf("dependencies only supported in appc builds")
+}
+func (a *ACBuild) AddIsolator(name string, value []byte) (err error) {
+	if err = a.lock(); err != nil {
+		return err
+	}
+	defer func() {
+		if err1 := a.unlock(); err == nil {
+			err = err1
+		}
+	}()
+	switch m := a.man.(type) {
+	case *appc.Manifest:
+		return m.AddIsolator(name, value)
+	}
+	return fmt.Errorf("dependencies only supported in appc builds")
+}
+func (a *ACBuild) RemoveIsolator(imageName string) (err error) {
+	if err = a.lock(); err != nil {
+		return err
+	}
+	defer func() {
+		if err1 := a.unlock(); err == nil {
+			err = err1
+		}
+	}()
+	switch m := a.man.(type) {
+	case *appc.Manifest:
+		return m.RemoveIsolator(imageName)
+	}
+	return fmt.Errorf("dependencies only supported in appc builds")
+}
+func (a *ACBuild) SetName(name string) (err error) {
+	if err = a.lock(); err != nil {
+		return err
+	}
+	defer func() {
+		if err1 := a.unlock(); err == nil {
+			err = err1
+		}
+	}()
+	switch m := a.man.(type) {
+	case *appc.Manifest:
+		return m.SetName(name)
+	}
+	return fmt.Errorf("setting names only supported in appc builds")
+}
+func (a *ACBuild) SetPostStop(exec []string) (err error) {
+	if err = a.lock(); err != nil {
+		return err
+	}
+	defer func() {
+		if err1 := a.unlock(); err == nil {
+			err = err1
+		}
+	}()
+	switch m := a.man.(type) {
+	case *appc.Manifest:
+		return m.SetPostStop(exec)
+	}
+	return fmt.Errorf("event handlers only supported in appc builds")
+}
+func (a *ACBuild) SetPreStart(exec []string) (err error) {
+	if err = a.lock(); err != nil {
+		return err
+	}
+	defer func() {
+		if err1 := a.unlock(); err == nil {
+			err = err1
+		}
+	}()
```

One option would be to silently drop errors from `a.unlock()`, but I dislike that for obvious reasons